### DISCUSSION TITLE
[BUG] Plotting on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by raceface1** [Chia-Network/chia-blockchain#3969] [Support]
Plotting on Raspberry Pi 4 fails every time after few hours, looks like plotting process gets killed 

Raspberry Pi 4 - 4GB version

Starting phase 2/4: Backpropagation into tmp files... Thu May  6 08:36:22 2021
Backpropagating on table 7
scanned table 7
scanned time =  596.860 seconds. CPU (29.230%) Thu May  6 08:46:21 2021
sorting table 7
Backpropagating on table 6
scanned table 6
scanned time =  1053.566 seconds. CPU (82.630%) Thu May  6 09:30:06 2021
sorting table 6
sort time =  4051.678 seconds. CPU (54.730%) Thu May  6 10:37:38 2021
Backpropagating on table 5
scanned table 5
scanned time =  862.661 seconds. CPU (95.830%) Thu May  6 10:52:05 2021
sorting table 5
sort time =  3705.220 seconds. CPU (55.990%) Thu May  6 11:53:50 2021
Backpropagating on table 4
scanned table 4
scanned time =  855.177 seconds. CPU (94.100%) Thu May  6 12:08:12 2021
sorting table 4
sort time =  3756.554 seconds. CPU (54.690%) Thu May  6 13:10:48 2021
Backpropagating on table 3
scanned table 3
scanned time =  842.955 seconds. CPU (95.810%) Thu May  6 13:24:53 2021
sorting table 3
sort time =  3680.357 seconds. CPU (56.140%) Thu May  6 14:26:14 2021
Backpropagating on table 2
scanned table 2
scanned time =  869.055 seconds. CPU (95.890%) Thu May  6 14:40:52 2021
sorting table 2
sort time =  3657.098 seconds. CPU (56.930%) Thu May  6 15:41:49 2021
table 1 new size: 3425196920
Time for phase 2 = 25536.406 seconds. CPU (60.440%) Thu May  6 15:41:59 2021
Wrote: 268

Starting phase 3/4: Compression from tmp files into "/media/Temp/plot-k32-2021-0                                                                                                                                                             5-05-11-44-74a627d1f68d9c33482f4fc3601577c04e037b9916e4c0291a4a3789e18bfc26.plot                                                                                                                                                             .2.tmp" ... Thu May  6 15:41:59 2021
Compressing tables 1 and 2
        Bucket 0 uniform sort. Ram: 1.894GiB, u_sort min: 1.250GiB, qs min: 0.31                                                                                                                                                             3GiB.
Killed

**Comment by chiabug** [comment]
Check the amount of space available in the temp folder.

**Comment by raceface1** [comment]
Temp partition has 4TB of space. So this is not an issue. Also when it fails noting is added to the log so it`s hard to debug.

**Reply by raceface1** [reply]
Noticed when using below plotting lasts 12-14h:
chia plots create -k 32 -b 2000

When using below plotting fails after 3-4h:
chia plots create -k 32 -b 4000

**Comment by rdooliveira** [comment]
Hello, 

I have faced the same issues on Linux, Ubuntu and on Windows.

After a few hours of creating just one plot: 

- Windows app closes automatic
- Ubuntu Desktop Gui also closes without any log.
- Ubuntu without Desktop, just CLI closes with folowing message (Killed)
- (chia plots create -k32 -n 100)

I have 2 TB on each hard drive, so there is plenty of space for just one plot.

Ubuntu
Forward propagation table time: 4655.660 seconds. CPU (126.580%) Fri May  7 16:10:23 2021
Computing table 6
        Bucket 0 uniform sort. Ram: 3.250GiB, u_sort min: 1.375GiB, qs min: 0.687GiB.
        Bucket 1 uniform sort. Ram: 3.250GiB, u_sort min: 1.375GiB, qs min: 0.687GiB.
        Bucket 2 uniform sort. Ram: 3.250GiB, u_sort min: 1.375GiB, qs min: 0.687GiB.
        Bucket 3 uniform sort. Ram: 3.250GiB, u_sort min: 2.750GiB, qs min: 0.688GiB.
Killed
(venv) XX@X:~/chia-blockchain$ 

My friends also claim that Chia on windows closes a lot when plotting and they have to try many times.

I think this is not a BUG, but a useful Feature for Chia .....  Is this feature for increasing Difficulty levels ?

Best regards

**Reply by rdooliveira** [reply]
Another test on another different machine.

        Bucket 56 uniform sort. Ram: 3.250GiB, u_sort min: 1.125GiB, qs min: 0.562GiB.
        Bucket 57 uniform sort. Ram: 3.250GiB, u_sort min: 2.250GiB, qs min: 0.563GiB.
Killed

**Reply by rdooliveira** [reply]
Another try

Bucket 125 uniform sort. Ram: 3.250GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
        Bucket 126 uniform sort. Ram: 3.250GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
        Bucket 127 uniform sort. Ram: 3.250GiB, u_sort min: 0.563GiB, qs min: 0.281GiB.
        Total matches: 4294914389
Forward propagation table time: 3220.137 seconds. CPU (131.500%) Fri May  7 19:53:04 2021
Computing table 3
        Bucket 0 uniform sort. Ram: 3.250GiB, u_sort min: 2.250GiB, qs min: 0.563GiB.
Killed

**Reply by raceface1** [reply]
Plotting never failed for me on Windows (created 20+ plots)

Also finally was sucesfull when plotting with below parameters:
chia plots create -k 32 -b 1000

when -b is set to anything higher than 1000 it fails

**Comment by sfxworks** [comment]
kernel kills memory hoarders if system is running low.
btw what times are you getting?

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
